scikit/notebooks/Scikit Learn on CMLE.ipynb

```diff
@@ -0,0 +1,702 @@
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 3881k  100 3881k    0     0  2733k      0  0:00:01  0:00:01 --:--:-- 2733k\n",
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 1956k  100 1956k    0     0  2034k      0 --:--:-- --:--:-- --:--:-- 2033k\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Create a directory to hold the data\n",
+    "! mkdir census_data\n",
+    "\n",
+    "# Download the data\n",
+    "! curl https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data --output census_data/adult.data\n",
+    "! curl https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.test --output census_data/adult.test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 1: Train/Save the model \n",
+    "First, the data is loaded into a numpy array that can be used by Scikit Learn. Then a simple model is created and fit against the training data. Lastly, sklearn's built in version of joblib is used to save the model to a file that can be uploaded to ML Engine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Done\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "from sklearn import svm\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.externals import joblib\n",
+    "\n",
+    "# Define the format of your input data including unused columns (These are the columns from the census data files)\n",
+    "COLUMNS = (\n",
+    "    'age',\n",
+    "    'workclass',\n",
+    "    'fnlwgt',\n",
+    "    'education',\n",
+    "    'education-num',\n",
+    "    'marital-status',\n",
+    "    'occupation',\n",
+    "    'relationship',\n",
+    "    'race',\n",
+    "    'sex',\n",
+    "    'capital-gain',\n",
+    "    'capital-loss',\n",
+    "    'hours-per-week',\n",
+    "    'native-country',\n",
+    "    'income-level'\n",
+    ")\n",
+    "\n",
+    "# Categorical columns are columns that need to be turned into a numerical value to be used by Scikit Learn\n",
+    "CATEGORICAL_COLUMNS = (\n",
+    "    'workclass',\n",
+    "    'education',\n",
+    "    'marital-status',\n",
+    "    'occupation',\n",
+    "    'relationship',\n",
+    "    'race',\n",
+    "    'sex',\n",
+    "    'native-country'\n",
+    ")\n",
+    "\n",
+    "\n",
+    "# Load the training census dataset\n",
+    "with open('./census_data/adult.data', 'r') as train_data:\n",
```

Comment:
This will read entire file in memory. You want to read in chunks, use chunksize.

Reply:
That would add a lot of complexity, such as training the model in chunks. Should we have a separate sample that is focused on showing how to train models based on large datasets? While this one is still focused on introducing sklearn online prediction.